spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -243,34 +243,88 @@
+        let(:responses) do
+          [
+            request_passed_proofing_results_args,
+            request_in_progress_proofing_results_args,
+            { status: 500 },
+            request_failed_proofing_results_args,
+            request_expired_proofing_results_args,
+          ]
+        end
+
+        before do
+          allow(InPersonEnrollment).to receive(:needs_usps_status_check).
+            and_return(pending_enrollments)
+          stub_request_proofing_results_with_responses(*responses)
+        end
 
-        job.perform(Time.zone.now)
+        it 'logs a message with counts of various outcomes when the job completes' do
+          job.perform(Time.zone.now)
 
-        expect(job_analytics).to have_logged_event(
-          'GetUspsProofingResultsJob: Job completed',
-          duration_seconds: anything,
-          enrollments_checked: 5,
-          enrollments_errored: 1,
-          enrollments_expired: 1,
-          enrollments_failed: 1,
-          enrollments_in_progress: 1,
-          enrollments_passed: 1,
-        )
+          expect(job_analytics).to have_logged_event(
+            'GetUspsProofingResultsJob: Job completed',
+            duration_seconds: anything,
+            enrollments_checked: 5,
+            enrollments_errored: 1,
+            enrollments_expired: 1,
+            enrollments_failed: 1,
+            enrollments_in_progress: 1,
+            enrollments_passed: 1,
+          )
 
-        expect(
-          job_analytics.events['GetUspsProofingResultsJob: Job completed'].
-            first[:duration_seconds],
-        ).to be >= 0.0
+          expect(
+            job_analytics.events['GetUspsProofingResultsJob: Job completed'].
+              first[:duration_seconds],
+          ).to be >= 0.0
+        end
+
+        it 'logs a message with counts of various outcomes when the job is terminated' do
+          error_message = 'too many!'
+          analytics_title = 'GetUspsProofingResultsJob: Job terminated'
+          # make a subclass of GetUspsProofingResultsJob that raises a
+          # ConcurrencyExceededError on its last enrollment
+          class ErrorJob < GetUspsProofingResultsJob
+            private
+
+            def check_enrollment(enrollment, is_not_last_enrollment)
+              if is_not_last_enrollment
+                super(enrollment, is_not_last_enrollment)
+              else
+                raise GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError,
+                      'too many!'
+              end
+            end
+          end
+
+          allow_any_instance_of(ErrorJob).to receive(:analytics).and_return(job_analytics)
+
+          # rescue so we can test for the error being re-raised
+          begin
+            ErrorJob.perform_now(Time.zone.now)
+          rescue GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError => err
+            expect(err.message).to eq(error_message)
+          end
```

Comment:
RSpec has matchers we can use here

```suggestion
          expect { ErrorJob.perform_now(Time.zone.now) }.
             to raise_error(GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError, error_message)
```

Reply:
Thank you, done in 19fd277a6b487d6d1c9dd3caceabd43e6941ceb0